src/com/palmergames/bukkit/towny/object/metadata/CustomDataField.java

```diff
@@ -89,11 +88,12 @@ public String toString() {
 	 * @param str - The metadata string to load
 	 * @return - The data field defined by the string
 	 */
+	@SuppressWarnings("unchecked")
     public static CustomDataField load(String str) {
         String[] tokens = str.split(",");
```

Comment:
With a generic API, especially one where keys are set by external developers (most likely not heavily familiar with the towny codebase), I think there needs to be inherent key and value checking, just in case a developer decides to create a key/value that contains the delimiter this load method uses (orthe separator that splits metadata values).

Reply:
Agreed